crates/matrix-sdk-sqlite/src/crypto_store.rs

```diff
@@ -499,6 +500,29 @@ trait SqliteObjectCryptoStoreExt: SqliteObjectExt {
             .await?)
     }
 
+    async fn mark_inbound_group_sessions_as_backed_up(&self, session_ids: Vec<Key>) -> Result<()> {
+        if session_ids.is_empty() {
+            // We are not expecting to be called with an empty list of sessions
+            warn!("No sessions to mark as backed up!");
+            return Ok(());
+        }
+
+        let session_ids_len = session_ids.len();
+
+        self.chunk_large_query_over(session_ids, None, move |session_ids| {
```

Comment:
@Hywan I've attempted to refactor it to use `chunk_large_query_over` - what do you think?

I couldn't find a way to do it in one transaction - I'm not sure whether that's needed or not.

I think I would need to add a new `with_transaction_sync` and that would need a new `interact_async` - do you agree?

Reply:
After discussing with @richvdh and @BillCarsonFr we think not having it all in a single transaction is OK.